src/workers/python/init.py

```diff
@@ -55,6 +56,8 @@ def cb(typ, dat, **kwargs):
                         raise ValueError(f"Unknown output part type {typ}")
             elif event_type == "input":
                 return cb("input", data["prompt"], contentType="text/plain")
+            elif event_type == "sleep":
+                return cb("sleep", str(data["seconds"]*1000), contentType="text/integer")
```

Comment:
I suggest using `makeRunnerCallback` from pyodide-worker-runner and moving this event translation logic to JS.

Reply:
I'm still unsure what the best approach is. The makeRunnerCallback method expects an array of callbacks, which becomes quite unwieldy. I moved to a PubSub system to prevent needing to pass many callbacks. I would think that adding a catch-all to makeRunnerCallback would make it a bit more user-friendly for this specific case, but I wouldn't call this part of my code stable yet.

Reply:
Here, I meant that makeRunnerCallback expects an object with at least 2 callbacks and then extra callbacks per possible type. I have an error callback, the interrupt just in case, and potentially a debugging callback which makes this unwieldy to pass around. If more types get added, this also poses issues for makeRunnerCallback. The translation in Python is not exactly ideal either, so I'll think about a hybrid approach to clean this up.

Reply:
It sounds like it would be better if makeRunnerCallback didn't try to be so helpful with other event types and just let you provide a function to act as a default. That way you could just let other events pass through to your own system.

Reply:
A catch-all would indeed be more practical in my opinion.